Fixes mentions of `contrib` and replaces with `main` (#7066)
umbraco/UmbracoDocs: contributing/backoffice-project/contributing.md

### Caveats

> The backoffice can be ran and tested against a real Umbraco instance by cloning down the Umbraco-CMS `main` branch, but there are no guarantees about how well it works yet.

Reviewer: ```suggestion
You can test the backoffice against a real Umbraco instance by cloning down the Umbraco-CMS `main` branch, but functionality is not guaranteed.
```
